src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -0,0 +1,271 @@
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        TransactionsModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  describe('GET multisig transactions by Safe', () => {
+    it('Failure: Config API fails', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      mockNetworkService.get.mockRejectedValueOnce({
+        status: 500,
+      });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/multisig-transactions`)
+        .expect(500)
+        .expect({
+          message: 'An error occurred',
+          code: 500,
+        });
+
+      expect(mockNetworkService.get).toBeCalledTimes(1);
+      expect(mockNetworkService.get).toBeCalledWith(
+        `${safeConfigApiUrl}/api/v1/chains/${chainId}`,
+        undefined,
+      );
+    });
+
+    it('Failure: Transaction API fails', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = chainFactory(chainId);
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.get.mockRejectedValueOnce({
+        status: 500,
+      });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/multisig-transactions`)
+        .expect(500)
+        .expect({
+          message: 'An error occurred',
+          code: 500,
+        });
+
+      expect(mockNetworkService.get).toBeCalledTimes(2);
+      expect(mockNetworkService.get).toBeCalledWith(
+        `${safeConfigApiUrl}/api/v1/chains/${chainId}`,
+        undefined,
+      );
+      expect(mockNetworkService.get).toBeCalledWith(
+        `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/multisig-transactions/`,
+        expect.objectContaining({
+          params: expect.objectContaining({
+            ordering: '-modified',
+            safe: safeAddress,
+            trusted: true,
+          }),
+        }),
+      );
+    });
+
+    it('Failure: data validation fails', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = chainFactory(chainId);
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.get.mockResolvedValueOnce({
+        data: { results: ['invalidData'] },
+      });
+
+      await request(app.getHttpServer())
```

Comment:
Do we need to await here if there is nothing after this call?

Reply:
I tried it in [9590b4e](https://github.com/5afe/safe-client-gateway-nest/pull/197/commits/9590b4edc09b406d88e253fcdf98e6725d60da84) and didn't work. I think not awaiting the Promise to resolve is letting Jest execute the `afterAll` hook and therefore closing down the app, I think that's the reason I was getting a 503 when removing the `await` instruction 🙂